libsolidity/interface/FileReader.cpp

```diff
@@ -153,7 +153,7 @@ ReadCallback::Result FileReader::readFile(string const& _kind, string const& _so
 			return ReadCallback::Result{false, "File outside of allowed directories."};
 
 		if (!boost::filesystem::is_regular_file(candidates[0]))
-			return ReadCallback::Result{false, "Not a valid file."};
+			return ReadCallback::Result{false, "Not a valid existing file: " + candidates[0].string()};
```

Comment:
@cameel as I'd like to get the actual PR (solc LSP functionality) merged, I'd say let's do this as part of the referenced issues. Would that be fine?

Reply:
Yeah, sure. As @chriseth has said, this is actually already covered by @joshuatarkwski's PR.

Reply:
non-LSP commits factored out into #12304 and I've dropped the commit on the other side. I'll touch this PR (and branch) not until being home because I apparently haven't pushed it all up last time :boom: :